src/hooks/useSetFocus.js

```diff
@@ -0,0 +1,34 @@
+import { useState, useLayoutEffect } from 'react'
+
+/**
+ * Hook sets focus on the passed ref. Will pass focus to the previously focused
+ * element when the component is unmounted
+ *
+ * @public
+ *
+ * @param {React.MutableRefObject<HTMLElement | null>} ref - the coponent's ref
+ * @param {boolean} isFocusReady - if true, the ref can be focused
+ *
+ * @return {boolean} - a boolean that represents if the component is currently focused
+ */
+function useSetFocus(ref, isFocusReady) {
```

Comment:
I don't see this Hook used anywhere, was that intentional?

Reply:
That's correct, I wrote this hook for a previous working version of the `modal` and is not used in the current implementation. 

I included the hook in this PR since it is a reusable and useful hook when focus needs to be conditionally.

Reply:
Yeah, good catch. Although I really like this hook, it might be best to wait for a use case 🤔 unless you can find a place in the codebase and redirect it to use this 😋 

I dunno, I'll just leave it up to Brad. Be nice to get feedback on the feedback though :)